runtime/src/comm/ofi/comm-ofi-oob-pmi2.c

```diff
@@ -290,53 +290,64 @@ void decode_kvs(char* raw, const char* enc, size_t size) {
   }
 }
 
-int chpl_comm_ofi_oob_locales_on_node(void) {
+typedef struct locale_info_t {
+  uint64_t  hash;     // hash of the locale's hostname
+  int       nodeID;   // locale's ID
+} locale_info_t;
+
+int chpl_comm_ofi_oob_locales_on_node(int *rank) {
```

Comment:
Want to chat about this code -- seems like we should be able to get this information from PMI, but I think we're computing it ourselves now

Reply:
Plan is to see if we can do this with PMI. May be some old context/info about how to do this in:
 - https://github.com/chapel-lang/chapel/pull/18279
 - https://github.com/chapel-lang/chapel/pull/18181

Reply:
Done.